ipn/ipnlocal/local.go

```diff
@@ -4791,3 +4798,23 @@ func (b *LocalBackend) ResetAuth() error {
 	}
 	return b.resetForProfileChangeLockedOnEntry()
 }
+
+// StreamDebugCapture writes a pcap stream of packets traversing
+// tailscaled to the provided response writer.
+func (b *LocalBackend) StreamDebugCapture(w http.ResponseWriter) error {
+	var s *capture.Sink
+
+	b.mu.Lock()
+	if b.debugSink == nil {
+		s = capture.New()
+		b.debugSink = s
+		b.e.InstallCaptureHook(s.LogPacket)
+	} else {
+		s = b.debugSink
+	}
+	b.mu.Unlock()
+
+	s.RegisterOutput(w)
+	s.Wait()
```

Comment:
you should take a context instead, then this becomes a select on both the context and the LocalBackend's shutdownCtx

Reply:
And after the context(s) are done, I'd prefer to see an explicit unregister call to stop the sink.

And why can't there be multiple going on at once?